modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessContextInfo.java

```diff
@@ -0,0 +1,173 @@
+import org.elasticsearch.painless.lookup.PainlessMethod;
+import org.elasticsearch.script.ScriptContext;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class PainlessContextInfo implements Writeable, ToXContentObject {
+
+    public static final ParseField NAME = new ParseField("name");
+    public static final ParseField CLASSES = new ParseField("classes");
+    public static final ParseField IMPORTED_METHODS = new ParseField("imported_methods");
+    public static final ParseField ClASS_BINDINGS = new ParseField("class_bindings");
+    public static final ParseField INSTANCE_BINDINGS = new ParseField("instance_bindings");
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<PainlessContextInfo, Void> PARSER = new ConstructingObjectParser<>(
+            PainlessContextInfo.class.getCanonicalName(),
+            (v) ->
+                    new PainlessContextInfo(
+                            (String)v[0],
+                            (List<PainlessContextClassInfo>)v[1],
+                            (List<PainlessContextMethodInfo>)v[2],
+                            (List<PainlessContextClassBindingInfo>)v[3],
+                            (List<PainlessContextInstanceBindingInfo>)v[4]
+                    )
+    );
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME);
+        PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(),
+                (p, c) -> PainlessContextClassInfo.fromXContent(p), CLASSES);
+        PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(),
+                (p, c) -> PainlessContextMethodInfo.fromXContent(p), IMPORTED_METHODS);
+        PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(),
+                (p, c) -> PainlessContextClassBindingInfo.fromXContent(p), ClASS_BINDINGS);
+        PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(),
+                (p, c) -> PainlessContextInstanceBindingInfo.fromXContent(p), INSTANCE_BINDINGS);
+    }
+
+    private final String name;
+    private final List<PainlessContextClassInfo> classes;
+    private final List<PainlessContextMethodInfo> importedMethods;
+    private final List<PainlessContextClassBindingInfo> classBindings;
+    private final List<PainlessContextInstanceBindingInfo> instanceBindings;
+
+    public PainlessContextInfo(ScriptContext<?> scriptContext, PainlessLookup painlessLookup) {
+        this(
+                scriptContext.name,
+                painlessLookup.getClasses().stream().map(
+                        javaClass -> new PainlessContextClassInfo(javaClass, painlessLookup.lookupPainlessClass(javaClass))
+                ).collect(Collectors.toList()),
+                painlessLookup.getImportedPainlessMethodsKeys().stream().map(importedPainlessMethodKey -> {
```

Comment:
maybe add `.sort(custom comparator)` to the chaining here in order to generate consistent response?
This will then also allow to the test the response in the yaml test?

Reply:
Sorted everything that made sense to sort.